packages/c/catch2/xmake.lua

```diff
@@ -29,11 +30,23 @@ package("catch2")
     end
 
     on_load(function (package)
+       if package:config("use_catch2main") then
+            package:add("components", "main")
+        end
+        package:add("components", "lib")
+    end)
+
+    on_component("main", function (package, component)
+        if package:version():ge("3.0") then
+            component:add("links", "Catch2Main")
+        end
+    end)
+
+    on_component("lib", function (package, component)
         if package:version():ge("3.0") then
-            package:add("deps", "cmake")
-            package:add("links", "Catch2Main", "Catch2")
+            component:add("links", "Catch2")
         else
-            package:set("kind", "library", {headeronly = true})
+            component:set("kind", "library", {headeronly = true})
```

Comment:
set_kind only for package, not component:set

Reply:
fixed